tests/queries/models.py

```diff
@@ -723,3 +724,9 @@ class Meta:
 class CustomDbColumn(models.Model):
     custom_column = models.IntegerField(db_column='custom_name', null=True)
     ip_address = models.GenericIPAddressField(null=True)
+
+
+class ReturningModel(models.Model):
+    # There's no official support for default=Now or returning=True. This
+    # is merely to test some internal APIs.
+    created = models.DateTimeField(default=Now, returning=True, editable=False)
```

Comment:
Can you think of a case where you'd want `default=Now(), returning=False`?

What happens if `returning=False`, does the field gets _deferred_ on creation? We certainly don't want `.created` to be a `Now` instance.

What if `returning=True` is passed and the backend doesn't support it? Should we attempt a `SELECT` to retrieve the value or does the field gets deferred.

These are questions that were [discussed in the past](https://github.com/django/deps/pull/36#issuecomment-261151212) but had no clear consensus and have to be tested before we can proceed with this PR.